userservice/src/main/java/com/infodation/userservice/services/UserServiceImpl.java

```diff
@@ -32,11 +38,42 @@
 public class UserServiceImpl implements IUserService {
     private static final Logger logger = LoggerFactory.getLogger(UserServiceImpl.class);
     private final UserRepository userRepository;
-    public UserServiceImpl(UserRepository userRepository) {
+    private final RestTemplate restTemplate;
+
+    public UserServiceImpl(UserRepository userRepository, RestTemplate restTemplate) {
         this.userRepository = userRepository;
+        this.restTemplate = restTemplate;
+    }
+
+    @Override
+    public TaskUserResponseDTO getUserWithTasks(String userId) {
+        // Find the user in the database based on user_id
+        User user = userRepository.findByUserId(userId).orElse(null);
+        checkUser(userId);
+        // Use UserMapper to convert User to UserDTO
+        UserDTO userDTO = UserMapper.INSTANCE.userToUserDTO(user);
+
+        ResponseEntity<List<TaskDTO>> taskResponse = restTemplate.exchange(
+                "http://localhost:8081/api/tasks/user/" + user.getId(),
+                HttpMethod.GET,
+                null,
+                new ParameterizedTypeReference<List<TaskDTO>>() {
+                }
+        );
```

Comment:
Please add `http://localhost:8081` as a configuration from the application properties.
In deployment, we will override this to another value via ENV var